onnxscript/ir/passes/common/common_subexpression_elimination_test.py

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import unittest
+
+import numpy as np
+import onnxruntime as ort
+
+from onnxscript import FLOAT, ir, script
+from onnxscript import opset18 as op
+from onnxscript.ir.passes.common import common_subexpression_elimination
+
+
+class TestCommonSubexpressionEliminationPass(unittest.TestCase):
+    def check_graph(self, model: ir.Model, inputs: list[ir.Value], delta_nodes: list[int]):
+        """Check if the model applied the CSE pass correctly.
+
+        Args:
+            model (ir.Model): The model to check.
+            inputs (list[ir.Value]): The inputs to the model.
+            delta_nodes (list[int]): The expected change in the number of nodes in the model.
+                                     The length of this list should match the number of graphs
+                                     in the model. (to support subgraphs in the future)
+
+        Raises:
+            AssertionError: If the model does not match the expected number of nodes or outputs.
+
+        """
+        assert len(list(model.graphs())) == len(delta_nodes)
+        # Log all results from the original model.
+        # 1. model graph node counts
+        original_graphs_node_count = np.array([graph.num_nodes() for graph in model.graphs()])
+        model_proto = ir.serde.serialize_model(model)
+
+        # 2. model outputs
+        ort_inputs = {
+            k.name: np.random.rand(*v.shape).astype(np.float32)
+            for k, v in zip(model.graph.inputs, inputs)
+        }
+        original_model_session = ort.InferenceSession(model_proto.SerializeToString())
+        original_model_results = original_model_session.run(None, ort_inputs)
+
+        result = common_subexpression_elimination.CommonSubexpressionEliminationPass()(model)
+
+        result_graphs_node_count = np.array([graph.num_nodes() for graph in model.graphs()])
+        # Check if the number of nodes in the model is correct
+        self.assertTrue(
+            np.array_equal(
+                original_graphs_node_count, np.add(result_graphs_node_count, delta_nodes)
+            )
+        )
+        self.assertEqual(
+            result.modified, any(original_graphs_node_count > result_graphs_node_count)
+        )
+
+        result_proto = ir.serde.serialize_model(result.model)
+        result_session = ort.InferenceSession(result_proto.SerializeToString())
+        result_results = result_session.run(None, ort_inputs)
+
+        # Check if the models produce the same output
+        # with the same inputs
+        for idx, original_model_result in enumerate(original_model_results):
+            np.testing.assert_allclose(
+                original_model_result, result_results[idx], rtol=1e-5, atol=1e-5
+            )
+
+    def test_two_branches_with_the_same_operations_is_csed(self):
+        """Test if two branches with the same operations are CSEd.
```

Comment:
nit: why do you say "branches"? Somehow it suggests an "If" node branches to me.